app.js

```diff
@@ -21,5 +21,43 @@ app.use(express.static(path.join(__dirname, 'public')));
 app.get('/', (req, res) => {
   res.render('index');
 });
+app.get('/beer', (req, res) => {
+    
+    
+  punkAPI.getBeers()
+  .then((beersFromApi) => { 
+    console.log(beersFromApi)
+    res.render('beer', {beersFromApi})
+  })
+  .catch(error => console.log(error));
+})
+;
+app.get('/random-beer', (req, res) => {
+
+  punkAPI
+  .getRandom()
+  .then(responseFromAPI => {
+    // your magic happens here
+    res.render('random-beer', {responseFromAPI});
+  })
+  .catch(error => console.log(error));
+
+});
+
+app.get('/beers/beer-:id', (req,res) => {
+
+  const id = req.params.id
+  punkAPI
+  .getBeer(id)
+  .then(each => {
+    res.render('random-beer', {each});
```

Comment:
Since you're reusing the `random-beer` page to render the beer details (which is correct) you need to pass the response from the API with the same name that you're referencing inside the `random-beer` page (responseFromAPI):

```
app.get('/beers/beer-:id', (req, res) => {
  const id = req.params.id;
  punkAPI
    .getBeer(id)
    .then(responseFromAPI => {
      res.render('random-beer', { responseFromAPI });
    })
    .catch(err => console.log(err));
});
```